File: scratch/search_oauth.js
const fs = require('fs');
const path = require('path');

const keywords = ["oauth/link", "oauth/callback", "clientId"];
const files = ["rpc_code.txt", "debug_rpc_search.txt", "discord-bot/index.js", "src/components/LoginModal.tsx"];

files.forEach(fn => {
    const filePath = path.join("e:/Code/CrunchyVerse", fn);
    if (fs.existsSync(filePath)) {
        console.log(`Searching in ${filePath}...`);
        const content = fs.readFileSync(filePath, 'utf8');
        keywords.forEach(kw => {
            let pos = -1;
            let count = 0;
            while ((pos = content.indexOf(kw, pos + 1)) !== -1) {
                count++;
                console.log(`  Found '${kw}' match ${count} at pos ${pos}:`);
                const start = Math.max(0, pos - 150);
                const end = Math.min(content.length, pos + 700);
                const snippet = content.substring(start, end);
                console.log(`-`.repeat(40));
                console.log(snippet);
                console.log(`-`.repeat(40));
            }
        });
    } else {
        console.log(`File not found: ${filePath}`);
    }
});

File: scratch/search_oauth.py
import re
import os

keywords = ["oauth/link", "oauth/callback", "clientId"]
files = ["rpc_code.txt", "debug_rpc_search.txt", "discord-bot/index.js", "src/components/LoginModal.tsx"]

for fn in files:
    path = os.path.join("e:/Code/CrunchyVerse", fn)
    if os.path.exists(path):
        print(f"Searching in {path}...")
        with open(path, "r", encoding="utf-8", errors="ignore") as f:
            content = f.read()
            for kw in keywords:
                matches = [m.start() for m in re.finditer(kw, content)]
                if matches:
                    print(f"  Found '{kw}' {len(matches)} times:")
                    for idx, pos in enumerate(matches):
                        start = max(0, pos - 200)
                        end = min(len(content), pos + 1000)
                        snippet = content[start:end]
                        print(f"    Match {idx + 1} at pos {pos}:\n{'-'*40}\n{snippet}\n{'-'*40}\n")
    else:
        print(f"File not found: {path}")
